spring-core/src/main/java/org/springframework/aot/hint/support/RuntimeHintsUtils.java

```diff
@@ -92,4 +97,25 @@ public static void registerAnnotationIfNecessary(RuntimeHints hints, MergedAnnot
 		}
 	}
 
+	/**
+	 * Register that the supplied resource should be made available at runtime.
+	 * <p>If the supplied resource is not a {@link ClassPathResource}, it will
+	 * not be registered.
+	 * @param hints the {@link RuntimeHints} instance to use
+	 * @param resource the resource to register
+	 * @throws IllegalArgumentException if the supplied resource does not
+	 * {@linkplain Resource#exists() exist}
+	 * @see ResourceHints#registerPattern(String)
+	 */
+	public static void registerResource(RuntimeHints hints, Resource resource) {
+		if (resource instanceof ClassPathResource classPathResource) {
+			Assert.isTrue(resource.exists(), () -> "Resource does not exist: " + resource);
```

Comment:
Couldn't this be `registerResourceIfNecessary` and then not register the resource if it's not found. That would be consistent with other methods in the same category.

Reply:
Yes, that's a good idea. 👍